selfdrive/ui/qt/maps/map.cc

```diff
@@ -253,6 +254,15 @@ void MapWindow::initializeGL() {
   m_map->setPitch(MIN_PITCH);
   m_map->setStyleUrl("mapbox://styles/commaai/clkqztk0f00ou01qyhsa5bzpj");
 
+  QObject::connect(m_map.data(), &QMapLibre::Map::mapLoadingFailed, [=](QMapLibre::Map::MapLoadingFailure err_code, const QString &reason) {
+    LOGE("Map loading failed with %d: '%s'\n", err_code, reason.toStdString().c_str());
+    // if time is wrong, re-create token and try connecting again in 5s
+    if (err_code == QMapLibre::Map::MapLoadingFailure::StyleLoadFailure && reason.contains("SSL handshake failed")) {
```

Comment:
checking an error string is generally fragile. style load failure might also just happen to be the first thing that MapLibre loads now, and it could easily change in the future

Reply:
This comes from mbgl: https://github.com/maplibre/maplibre-native-qt/blob/3a83b8ecfd961983cf3670a145ef6176333dd443/src/core/map_observer.cpp#L59-L60

Which is this section: https://github.com/mapbox/mapbox-gl-native/blob/f2778251c97ba3403582b9c04290c50f927fd338/src/mbgl/map/map_impl.cpp#L98-L100

So this is the source: https://github.com/mapbox/mapbox-gl-native/blob/f2778251c97ba3403582b9c04290c50f927fd338/src/mbgl/style/style_impl.cpp#L71-L75

So checking for the error code should be safe as we only load the style once, but we can remove the reason check which will just open us up to retrying for 404/403, etc.